src/back/space/src/api/space.cpp

```diff
@@ -0,0 +1,219 @@
+#include "space.hpp"
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+#include "../../../shared/errors.hpp"
+#include "../model/space_serialize.hpp"
+
+namespace svetit::space::handlers {
+
+Space::Space(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Space::HandleRequestJsonThrow(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body,
+	server::request::RequestContext&) const
+{
+	switch (req.GetMethod()) {
+		case server::http::HttpMethod::kGet:
+			return Get(req, body);
+		case server::http::HttpMethod::kPost:
+			return Post(req, body);
+		case server::http::HttpMethod::kDelete:
+			return Delete(req, body);
+		case server::http::HttpMethod::kHead:
+			return Head(req, body);
+		default: break;
+	}
+
+	formats::json::ValueBuilder res;
+	res["err"] = "Unsupported";
+	req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+	return res.ExtractValue();
+}
+
+formats::json::Value Space::Get(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body) const
+{
+	formats::json::ValueBuilder res;
+
+	try {
+		const auto& userId = req.GetHeader(headers::kUserId);
+		if (userId.empty())
+			throw errors::Unauthorized{};
+
+		const auto& id = req.GetArg("id");
+		const auto& key = req.GetArg("key");
+		const auto& link = req.GetArg("link");
+
+		if (id.empty() && key.empty() && link.empty())
+			throw errors::BadRequest{"No arguments"};
+
+		if (
+			(!id.empty() && (!key.empty() || !link.empty()))
+			|| (!key.empty() && (!id.empty() || !link.empty()))
+			|| (!link.empty() && (!id.empty() || !key.empty()))
+		)
+			throw errors::BadRequest{"Only one argument should be set"};
+
+		int method;
+
+		if (!id.empty()) {
+			method = 1;
+		} else if (!key.empty()) {
+			// todo - rewtite this, because key with UUID (that equals userId) inside could exists, it will not pass standard key regex
+			if (!_s.CheckKeyByRegex(key))
+				throw errors::BadRequest{"Key must be valid"};
+			method = 2;
+		} else if (!link.empty()) {
+			method = 3;
+		}
+
+		switch (method) {
+			case 1:
+				res = _s.GetById(id, userId);
+				break;
+			case 2:
+				res = _s.GetByKey(key, userId);
+				break;
+			case 3:
+				res = _s.GetByLink(link);
+				break;
+		}
```

Comment:
Кажется тут не произойдёт ничего страшного если мы обработаем эти параметры по очереди?
```suggestion
		if (req.HasArg("id"))
		{
			const auto id = utils::BoostUuidFromString(req.GetArg("id"));
			res = _s.GetById(id, userId);
		}
		else if (req.HasArg("key"))
		{
			const auto& key = req.GetArg("key");
			if (key != userId && !_s.CheckKeyByRegex(key))
				throw errors::BadRequest{"Key must be valid"};
			res = _s.GetByKey(key, userId);
		}
		else if (req.HasArg("link"))
		{
			const auto linkId = utils::BoostUuidFromString(req.GetArg("link"));
			res = _s.GetByLink(id, linkId);
		}
		else
			throw errors::BadRequest{"No arguments"};
```

Reply:
Готово.